### ISSUE TITLE
Custom tags?

### ISSUE THREAD

**Issue by AtlasJan** [photostructure/exiftool-vendored.js#55]
Hi, is there anything about adding custom tag definitions?

I have a set of custom XMP tags I use for catagorising images, is there a way to add their definitions so I can read them?

**If so:**

Could the documentation be updated to make this feature's use clearer if it's easy? and could you tell me how to do it?

**If not:**

I would like to see this feature implemented somehow. It might be a bit outside my wheelhouse, so I might not it myself.

**Comment by mceachen** [comment]
No tag values are removed by this library. 

If ExifTool extracts the tag and value, you'll get the result. 

In other words, just because something isn't in the `Tag` interface **doesn't mean the tags won't be passed back to you.**

Please run `exiftool -j $youfilename` and see if ExifTool knows how to extract that XMP tag. If it can't, this library can't either.

If you find something else going on, please provide a sample file that has those tags, and what you'd expect to see from `ExifTool.read()`.

**Comment by AtlasJan** [comment]
There's a bug in my initial statement, let me rephrase that.

Do I need to do anything in order to read namespaces I've defined in my
main exiftool config?

The namespace I want to read is defined in my config as follows:

|%Image::Exiftool::UserDefined = ( 'Image::ExifTool::XMP::Main' => {
galex => { SubDirectory => { TagTable =>
'Image::ExifTool::UserDefined::galex' }} } );
%Image::ExifTool::UserDefined::galex = ( GROUPS => { 0 => 'XMP', 1 =>
'XMP-galex', 2 => 'Image' }, NAMESPACE => { 'galex' =>
'http://atlasjan.sdf.org/xml/galex.xsd' }, WRITABLE => 'string', #
default to string-type tags Include => { Writable => 'boolean' },
Explicit => { Writable => 'boolean' }, Spoiler => { Writable =>
'boolean' }, Comment => { Writable => 'string' }, ); |

I've also attached it, just in case.

**Comment by AtlasJan** [comment]
There's a bug in my initial statement, let me rephrase that.

Do I need to do anything in order to read namespaces I've defined in my
main exiftool config?

The namespace I want to read is defined in my config as follows:

```perl
%Image::Exiftool::UserDefined = (
  'Image::ExifTool::XMP::Main' => {
    galex => { SubDirectory => { TagTable => 'Image::ExifTool::UserDefined::galex' }}
  }
);

%Image::ExifTool::UserDefined::galex = (
    GROUPS => { 0 => 'XMP', 1 => 'XMP-galex', 2 => 'Image' },
    NAMESPACE => { 'galex' => 'http://atlasjan.sdf.org/xml/galex.xsd' },
    WRITABLE => 'string', # default to string-type tags
    Include => { Writable => 'boolean' },
    Explicit => { Writable => 'boolean' },
    Spoiler => { Writable => 'boolean' },
    Comment => { Writable => 'string' },
);

```

**Comment by AtlasJan** [comment]
### Clarification:

As defining custom XMP tags and namespaces in ExifTool is done via user-defined tags in a [config file](https://stackoverflow.com/questions/24416653/is-it-possible-to-create-a-custom-namespace-tag-in-xmp-dublin-core-metadata#24430309), there is nothing in the documentation to suggest that I need to to the same by some mechanism, or any apparent method to import a config file into the vendored program.

I'm moaning in issues because I want the documentation for this process to be more readily apparent somehow, if it's implemented, if it's not implemented, I'd like it to be.

**Comment by mceachen** [comment]
Config files for the current release will work if they are in your home directory.

I'll release a new version tomorrow that adds support for setting the `EXIFTOOL_HOME` environment variable, or setting it in a new `exiftoolArgs` ExifTool constructor option.

Could you add documentation where you think it would be best? That'd be a great help.

**Comment by AtlasJan** [comment]
I guess I could? I'll work on a pull request or something, just tell me
what system out of the two you're going for.

**Comment by AtlasJan** [comment]
Looking over the commit, you seem to have done a better job than me at
adding documentation.


Nice work! Thanks for being patient with me stumbling over explaining
myself.

.
